include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1779,44 +1796,72 @@ namespace internal
               {
-                const Quadrature<1> dummy_quadrature(
-                  std::vector<Point<1>>(1, Point<1>()));
-                internal::MatrixFreeFunctions::ShapeInfo<Number> shape_info;
-                shape_info.reinit(dummy_quadrature,
-                                  dof_handler_fine.get_fe(
-                                    fe_index_pair.first.second),
-                                  0);
-                lexicographic_numbering_fine[fe_index_pair.second] =
-                  shape_info.lexicographic_numbering;
-
-                shape_info.reinit(dummy_quadrature,
-                                  dof_handler_coarse.get_fe(
-                                    fe_index_pair.first.first),
-                                  0);
-                lexicographic_numbering_coarse[fe_index_pair.second] =
-                  shape_info.lexicographic_numbering;
+                if (reference_cell == ReferenceCells::get_hypercube<dim>())
+                  {
+                    const Quadrature<1> dummy_quadrature(
+                      std::vector<Point<1>>(1, Point<1>()));
+                    internal::MatrixFreeFunctions::ShapeInfo<Number> shape_info;
+                    shape_info.reinit(dummy_quadrature,
+                                      dof_handler_fine.get_fe(
+                                        fe_index_pair.first.second),
+                                      0);
+                    lexicographic_numbering_fine[fe_index_pair.second] =
+                      shape_info.lexicographic_numbering;
+
+                    shape_info.reinit(dummy_quadrature,
+                                      dof_handler_coarse.get_fe(
+                                        fe_index_pair.first.first),
+                                      0);
+                    lexicographic_numbering_coarse[fe_index_pair.second] =
+                      shape_info.lexicographic_numbering;
+                  }
+                else
+                  {
+                    const auto dummy_quadrature =
+                      reference_cell.template get_gauss_type_quadrature<dim>(1);
+
+                    internal::MatrixFreeFunctions::ShapeInfo<Number> shape_info;
+                    shape_info.reinit(dummy_quadrature,
+                                      dof_handler_fine.get_fe(
+                                        fe_index_pair.first.second),
+                                      0);
+                    lexicographic_numbering_fine[fe_index_pair.second] =
+                      shape_info.lexicographic_numbering;
+
+                    shape_info.reinit(dummy_quadrature,
+                                      dof_handler_coarse.get_fe(
+                                        fe_index_pair.first.first),
+                                      0);
+                    lexicographic_numbering_coarse[fe_index_pair.second] =
+                      shape_info.lexicographic_numbering;
+                  }
```

Comment:
I don't see how this `else` clause differs from the other places, except that it uses a different dummy quadrature. But we never use anything from the quadrature (as it is a dummy), so I don't see why we can't use the one from the `if` clause also for non-hypercubes. Again, not the change in question here, but maybe we can fix it. If not, I would prefer to only set the quadrature formula in an `if`/`else` setup.